### DISCUSSION TITLE
SteamUnifiedMessages confusion

### DISCUSSION THREAD

**Discussion by NemuruYama** [SteamRE/SteamKit#1472] [General]
I am currently trying to use SteamUnifiedMessages to subscribe to a workshop item, and I was wondering how I am supposed to use it.

I currently have:
```cs
var steamUnifiedMessages = steamClient.GetHandler<SteamUnifiedMessages>();
if (steamUnifiedMessages is null)
{
    return;
}
var subscribeRequest = new CPublishedFile_Subscribe_Request
{
    publishedfileid = 3337263133,
    appid = 294100,
};
var subscribeResponse = await steamUnifiedMessages
    .SendMessage<CPublishedFile_Subscribe_Request, CPublishedFile_Subscribe_Response>(
        "PublishedFile.Subscribe",
        subscribeRequest
    );
```

But it throws an error that the task was cancelled.
Do I have the wrong Service/Rpc name or what is going on here?

**Comment by xPaw** [comment]
I think you're missing #1 from the name, but try using the direct generated call instead of using Send message manually. You also need to create a service. See the unified messages sample.

**Reply by NemuruYama** [reply]
> I think you're missing #1 from the name, but try using the direct generated call instead of using Send message manually. You also need to create a service. See the unified messages sample.

Thank you for the quick reply, that now works.
```cs
var service = steamUnifiedMessages.CreateService<PublishedFile>();
SteamUnifiedMessages.ServiceMethodResponse<CPublishedFile_Subscribe_Response>? resp = null;
try
{
    resp = await service.Subscribe(new CPublishedFile_Subscribe_Request { publishedfileid = 3337263133});
}
catch (Exception e)
{
    Console.WriteLine(e);
}
```

But there is still a problem, the subscribe call does literally nothing.
It returns an "OK", but it does not subscribe.

![image](https://github.com/user-attachments/assets/f1c5f88a-6bc3-49fd-bb77-b14c2470d566)

Edit: I also tested Unsubscribe and it has the same issue, where it does not unsubscribe from the workshop item even though it returns an "Ok" status.
